### PR TITLE
[New Rule] User Added as Owner for Azure Service Principal

### LINKED ISSUE #113: [New Rule] User Added as Owner for Azure Service Principal
## Description

A service principal in Azure is the application instance of a global application object in a single tenant or directory. A service principal is created in each tenant where the application is used and references the globally unique application object.

The service principal object defines what the application can do in the specific tenant, who can access the application, and what resources the app can access. A service principal object is created when an application is given permission to access resources in a tenant.

An adversary may add a user account as an owner for a service principal and use that account in order to define what an application can do in the Azure AD tenant.

```
event.module:azure and event.dataset:azure.auditlogs and event.category:AuditLogs and azure.auditlogs.operation_name:("Add owner to service principal") and event.outcome:Success
```

## Required Info

- **Eventing Sources:**

`filebeat-*`

- **Target Operating Systems:**

- **Platforms**

Azure

- **Target ECS Version:** `1.5.0`
- **New fields required in ECS for this?** No
- **Related issues or PRs** None

## Optional Info
- **References:**
  - https://docs.microsoft.com/en-us/azure/active-directory/develop/app-objects-and-service-principals

## Example Data

image

### PR DESCRIPTION
**Pull request by threat-punter**
## Issues

Resolves #113 

## Summary
Identifies when a user is added as an owner for an Azure service principal. The service principal object defines what the application can do in the specific tenant, who can access the application, and what resources the app can access.

A service principal object is created when an application is given permission to access resources in a tenant. An adversary may add a user account as an owner for a service principal and use that account in order to define what an application can do in the Azure AD tenant.

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

File: rules/azure/defense_evasion_user_added_as_owner_for_azure_application.toml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    "https://docs.microsoft.com/en-us/azure/active-directory/develop/app-objects-and-service-principals",
+]
+risk_score = 21
+rule_id = "38e5acdd-5f20-4d99-8fe4-f0a1a592077f"
+severity = "low"
+tags = ["Elastic", "Azure", "Continuous Monitoring", "SecOps", "Configuration Audit"]
+type = "query"
+
+query = '''
+event.module:azure and event.dataset:azure.auditlogs and event.category:AuditLogs and azure.auditlogs.operation_name:("Add owner to service principal") and event.outcome:Success
```
Comment (brokensound77) [inline rules/azure/defense_evasion_user_added_as_owner_for_azure_application.toml:32]: nit:

```suggestion
event.module:azure and event.dataset:azure.auditlogs and event.category:AuditLogs and azure.auditlogs.operation_name:"Add owner to service principal" and event.outcome:Success
```

Comment (threat-punter) [reply rules/azure/defense_evasion_user_added_as_owner_for_azure_application.toml:32]: 👍

**Review by Samirbous — APPROVED** [review]

File: rules/azure/defense_evasion_user_added_as_owner_for_azure_application.toml [hunk 2]
```diff
@@ -0,0 +1,33 @@
+[metadata]
+creation_date = "2020/08/20"
+ecs_version = ["1.5.0"]
```
Comment (peasead) [inline rules/azure/defense_evasion_user_added_as_owner_for_azure_application.toml:3]:
```suggestion
ecs_version = ["1.6.0"]
```

**Comment by threat-punter** [comment]
> Maybe `Account Manipulation`? Otherwise, looks good.

Done. Thanks!

**Comment by threat-punter** [comment]
> > Maybe `Account Manipulation`? Otherwise, looks good.
> 
> Done. Thanks!

Looks like I had a conflict and didn't realize that the ATT&CK metadata didn't get committed. I'll open a PR to fix that.

**Pull request merged**